internal/pkg/server/server.go

```diff
@@ -0,0 +1,92 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/frain-dev/convoy"
+	"github.com/frain-dev/convoy/config"
+	"github.com/frain-dev/convoy/internal/pkg/middleware"
+	"github.com/frain-dev/convoy/util"
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/render"
+	log "github.com/sirupsen/logrus"
+)
+
+type Server struct {
+	s *http.Server
+	M *middleware.Middleware
+}
+
+type CreateServer struct {
+	Cfg config.Configuration
+}
+
+func NewServer(c *CreateServer) *Server {
```

Comment:
We can refactor this further to use the exact dependency required: `func NewServer(port int) *Server`